File: common/ligands.R
# common/ligands.R — ligand panel metadata (single source: common/ligands.csv)
#
# Columns: ligand, label, class, color, lty, shape, trap_potency.
# trap_potency is the literature trapping potency relative to olaparib
# (empty = not determined; see data/01_curated/trapping_potency.csv).

load_ligands <- function() {
  args <- commandArgs(trailingOnly = FALSE)
  m <- grep("^--file=", args)
  if (length(m)) {
    script_dir <- dirname(normalizePath(sub("^--file=", "", args[m[1]])))
    p <- file.path(script_dir, "..", "common", "ligands.csv")
  } else {
    p <- "common/ligands.csv"
  }
  if (!file.exists(p)) {
    stop("cannot find common/ligands.csv (run scripts from the repository root)")
  }
  lig <- read.csv(p, stringsAsFactors = FALSE, na.strings = "")
  lig
}

File: common/ligands.py
"""Ligand panel metadata (single source: common/ligands.csv).

Columns: ligand, label, class, color, lty, shape, trap_potency
(trap_potency = trapping relative to olaparib; None = not determined).
"""
import csv
from pathlib import Path

CSV = Path(__file__).with_name("ligands.csv")


def load():
    with CSV.open(newline="") as fh:
        rows = []
        for r in csv.DictReader(fh):
            r["trap_potency"] = float(r["trap_potency"]) if r["trap_potency"] else None
            r["shape"] = int(r["shape"])
            rows.append(r)
        return rows


LIGANDS = {r["ligand"]: r for r in load()}


def colors():
    return {lig: r["color"] for lig, r in LIGANDS.items()}
